agent/tool-scripts/datalog/haproxy-ocp-datalog

```diff
@@ -399,7 +397,7 @@ while not terminate:
     next_start = time_stamp + interval
 
 # First wait on any existing sub-processes to complete.
-for router in _routers.values():
+for router in _routers.values():  # lgtm [py/unreachable-statement]
```

Comment:
So is this because LGTM can't tell that the previous loop terminates based on a global presumably set somewhere else? If so, seems like this deserves a bit of a comment to clarify, because out of the blue it looks weird. (And apparently there are a series of related changes; which one might take as emphasis on the generic philosophy that "globals are bad", I suppose... although the failure of analysis tools to deal with a language feature gracefully isn't actually quite the same as an argument that the feature is bad ...)

Reply:
I'm willing to cut LGTM some slack:  figuring out that the previous loop terminates due to an external, asynchronously generated signal is a bit of a tall order.  (But, I did report it to the LGTM folks, and they did agree that it was a false positive...but they aren't sure when they will be able to fix it -- they are working on security stuff, currently, so this issue went into the backlog.)

I wrestled with adding an additional comment but decided that that would only make the wart bigger.  I concluded that the presence of the `lgtm` token was enough of a clue, and, if someone really wanted to know what was going on, they could remove the directive and apply LGTM to the result.

I concur that, in general, globals are bad (and, I'll point out that the `global` statement in Python doesn't quite mean what it seems to say), but I'm not sure it would really be a win to wrap the `terminate` flag in a class with setters and getters so that it's obvious how it's being accessed...in particular, because LGTM would presumably _still_ have problems knowing when/how the setter is being called....